pylint/checkers/refactoring.py

```diff
@@ -261,12 +265,30 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
+        for part in node.body:
+            if isinstance(node.iter, astroid.node_classes.Name):
```

Comment:
node classes are exposed in global namespace, conventionally these exposed names are used - `astroid.Name`in this case, any other node type in other lines.